### PR TITLE
Item repository4

### PR DESCRIPTION
**Pull request by LawrenceWhalen**

### CODE REVIEW

File: lib/item_repository.rb [hunk 1]
```diff
@@ -23,20 +24,57 @@ def find_by_name(name)
...
   end
+
+  def find_all_by_merchant_id(merchant_id)
+    @csv_array.find_all do |item|
+      item.merchant_id == merchant_id
+    end
+  end
+
+  def max_id_number_new
+    old_max = @csv_array.max_by do |item|
```
[the remark this answers is not in the document]
Comment (LawrenceWhalen) [reply lib/item_repository.rb:51]: NO

[the remark this answers is not in the document]
Comment (LawrenceWhalen) [reply lib/item_repository.rb:51]: NO

**Comment by EmmyMorris** [comment]
I think this all looks awesome! Great job Lawrence!

File: spec/item_repository_spec.rb [hunk 2]
```diff
@@ -144,5 +144,95 @@
...
+        se = SalesEngine.from_csv(
+          :items     => './data/items.csv',
+          :merchants => './data/merchants.csv'
+        )
+
+        ir = se.items
+
+        expect(ir.find_all_by_price_in_range('100'..'1000')[5].name).to eq('Two tone blue stoneware pot')
+      end
+      it 'returns an empty array when no items are found' do
```
Comment (branfull) [inline spec/item_repository_spec.rb:159]: Starting with the 'end' on line 147 through the rest of the file, it looks like everything is indented by 2 extra spaces.

**Comment by branfull** [comment]
If you want to update the indents you can, but it looks like there is an extra 'end' at the very end of the item_repository_spec file

**Pull request merged**